user-interface/src/components/CaseDetail.test.tsx

```diff
@@ -78,7 +87,122 @@ describe('Case Detail screen tests', () => {
-        expect(debtorName).toHaveTextContent(testCaseDetail.debtorName);
+        expect(debtorName).toHaveTextContent(testCaseDetail.debtor.name);
+
+        const properties: Array<keyof Debtor> = ['address1', 'address2', 'address3', 'address4'];
+        properties.forEach((property) => {
+          const testId = `case-detail-debtor-${property}`;
+          if (testCaseDetail.debtor[property]) {
+            const element = screen.getByTestId(testId);
+            expect(element.innerHTML).toEqual(testCaseDetail.debtor[property]);
+          } else {
+            const element = screen.queryByTestId(testId);
+            expect(element).not.toBeInTheDocument();
+          }
+        });
+      },
+      { timeout: 5000 },
+    );
+  }, 20000);
+
+  const testCases = [
+    [undefined, undefined, undefined, undefined],
+    ['123 Rabbithole Lane', 'Unit 321', undefined, 'Ciudad Obregón GR 25443, MX'],
+    ['123 Rabbithole Lane', undefined, 'Unit 456', 'Ciudad Obregón GR 25443, MX'],
+    ['123 Rabbithole Lane', undefined, undefined, 'Ciudad Obregón GR 25443, MX'],
+    ['123 Rabbithole Lane', 'Unit', '111', 'Ciudad Obregón GR 25443, MX'],
+    ['123 Rabbithole Lane', 'Ciudad Obregón GR 25443, MX', undefined, undefined],
+    ['123 Rabbithole Lane', undefined, undefined, undefined],
+  ];
+
+  test.each(testCases)(
+    'should display debtor address with missing address2 and address3 for the case',
+    async (
+      address1: MaybeString,
+      address2: MaybeString,
+      address3: MaybeString,
+      address4: MaybeString,
+    ) => {
+      const testCaseDetail: CaseDetailType = {
+        caseId: caseId,
+        chapter: '15',
+        caseTitle: 'The Beach Boys',
+        dateFiled: '01-04-1962',
+        judgeName: rickBHartName,
+        closedDate: '01-08-1963',
+        dismissedDate: '01-08-1964',
+        assignments: [brianWilsonName, carlWilsonName],
+        debtor: {
+          name: 'Roger Rabbit',
+          address1,
+          address2,
+          address3,
+          address4,
+        },
+      };
+      render(
+        <BrowserRouter>
+          <CaseDetail caseDetail={testCaseDetail} />
+        </BrowserRouter>,
+      );
+
+      await waitFor(
+        async () => {
+          const properties: Array<keyof Debtor> = ['address1', 'address2', 'address3', 'address4'];
+          properties.forEach((property) => {
+            const testId = `case-detail-debtor-${property}`;
+            if (testCaseDetail.debtor[property]) {
+              const element = screen.getByTestId(testId);
+              expect(element.innerHTML).toEqual(testCaseDetail.debtor[property]);
+            } else {
+              const element = screen.queryByTestId(testId);
+              expect(element).not.toBeInTheDocument();
+            }
+          });
+        },
+        { timeout: 5000 },
+      );
+    },
+    20000,
+  );
+
+  test('should display debtor address with missing address3 for the case', async () => {
```

Comment:
Should we just remove this test because this case is covered by test above.

Reply:
I would be fine with that. We certainly don't want to test all combinations of presence/absence of each of four items, but it's hard to know whether any given combination will catch some future edge case regression.